### PR TITLE
nested schema issue fix

### PR DESCRIPTION
**Pull request by yug095**
fix the lira:
https://issues.cask.co/browse/HYDRA-91

http://builds.cask.co/browse/HYP-BAD26-2

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToAvroTransformer.java [hunk 1]
```diff
@@ -63,7 +63,12 @@ public GenericRecord transform(StructuredRecord structuredRecord) throws IOExcep
       if (schemaField == null) {
         throw new IllegalArgumentException("Input record does not contain the " + fieldName + " field.");
       }
-      recordBuilder.set(fieldName, convertField(structuredRecord.get(fieldName), schemaField.getSchema()));
+      if (schemaField.getSchema().getType() == co.cask.cdap.api.data.schema.Schema.Type.RECORD) {
```
Comment (albertshau) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToAvroTransformer.java]: this is not a good way to fix it. convertField already has a switch statement on the field type, you shouldn't duplicate that logic here, and shouldn't be creating a new transformer for each recursive call.

Instead, just remove the outputAvroSchema variable completely. The avro schema should always be calculated from the  cdap schema.

**Comment by albertshau** [comment]
lgtm assuming tests pass. Have you verified that the pipeline in the jira works now?

**Comment by yug095** [comment]
@albertshau Yes, it works now.

**Comment by yug095** [comment]
Bamboo Build Green, will merge it.

**Pull request merged**